pkg/kapp/cmd/app/delete.go

```diff
@@ -151,6 +153,10 @@ func (o *DeleteOptions) Run() error {
 
 	touch := ctlapp.Touch{App: app, Description: "delete", IgnoreSuccessErr: true}
 
+	if o.DeleteFlags.DisableCheckingFullDeletion {
+		shouldFullyDeleteApp = false
```

Comment:
This will prevent the app from being deleted, i.e. the metadata configmap that kapp creates for every app will not be deleted in this case. `shouldFullyDeleteApp` is used in cases of resource filtering, for example when we use filter in the `kapp delete` command to delete only a specific resource.
We would actually want to disable [this](https://github.com/vmware-tanzu/carvel-kapp/blob/b4115e24abd119a6b1dbe43080ec45a6b753ec46/pkg/kapp/app/labeled_app.go#L61) check so that we don't check if the resources are deleted or not before deleting the app itself.

Reply:
Thanks for the correction. And I think this is an common functionality for App, so I changed the App interface.